lib/streamlit/web/server/browser_websocket_handler.py

```diff
@@ -133,6 +133,11 @@ def open(self, *args, **kwargs) -> Awaitable[None] | None:
     def on_close(self) -> None:
         if not self._session_id:
             return
+
+        if not self.close_code or self.close_code not in [1000, 1001]:
+            # Unclean close. We may have already reconnected so do not disconnect.
+            return
+
```

Comment:
Hm, I don't think we want to do this here. This will cause sessions with websockets that disconnect uncleanly to potentially never be cleaned up and cause a memory leak.

Also, I think the change within `websocket_session_manager.py` alone should be sufficient to make our reconnect behavior more resilient.